src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -385,6 +384,11 @@ internal void SerializeEncoding(SerializationInfo info, StreamingContext context
             }
         }
 
+        // On Desktop, encoding instances that aren't cached in a static field are cached in
+        // a hash table by codepage.
+        private static volatile Hashtable encodings;
```

Comment:
I didn't really want to significantly change the desktop implementation as part of this PR, hence I left the existing `Hashtable`-based code mostly as-is.